src/hooks/project-hooks.tsx

```diff
@@ -178,16 +184,25 @@ export const ProjectProvider = ({
           loadDataset(gestureData);
           navigate(createDataSamplesPageUrl());
         } else {
-          // TODO: complain to the user!
+          setPostImportDialogState(PostImportDialogState.Error);
         }
       } else if (fileExtension === "hex") {
-        driverRef.current!.importFile({
-          filename: file.name,
-          parts: [await readFileAsText(file)],
-        });
+        const hex = await readFileAsText(file);
+        const makeCodeMagicMark = "41140E2FB82FA2BB";
+        // Check if is a MakeCode hex, otherwise show error dialog.
+        if (hex.includes(makeCodeMagicMark)) {
```

Comment:
Logic taken from https://github.com/microbit-foundation/microbit-android/blob/d1a49f745d1917d3d4e8de519e4deba9aa62c950/app/src/main/java/com/samsung/microbit/ui/activity/ProjectActivity.java#L2583-L2590

Reply:
Not sure if I should leave it as a comment in the actual code 🤔

Reply:
No that's fine.